compiler/rustc_codegen_ssa/src/traits/backend.rs

```diff
@@ -54,6 +55,10 @@ pub trait CodegenBackend {
     fn print_passes(&self) {}
     fn print_version(&self) {}
 
+    /// If this plugin provides additional builtin targets, provide them here.
+    /// Be careful: this is called *before* init() is called.
+    fn target_override(&self, opts: &config::Options) -> Option<Target>;
```

Comment:
The documentation says `targets`, but the return type is only a single target and the function name is `target_override`. I think you'll want to update the docs to replace `provide them here` with `provide the one enabled by the options here`?

Reply:
Also for convenience this could have a default impl of `None` so only the backends that use it have to provide it.